src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1173,6 +1190,10 @@ def rados_config_name(self):
         # type: () -> str
         return 'conf-' + self.service_name()
 
+    def validate(self) -> None:
+        if self.virtual_ip and (self.ip_addrs or self.networks):
+            raise SpecValidationError(f"Invalid NFS spec: Can not have both VIP and {'ip_addr' if self.ip_addrs else 'networks'}")
```

Comment:
```suggestion
            raise SpecValidationError(f"Invalid NFS spec: Cannot set virtual_ip and {'ip_addr' if self.ip_addrs else 'networks'} fields")
```
NIT, I think it's better to refer to the exact field name